pkg/downloader/manager.go

```diff
@@ -735,7 +744,13 @@ func (m *Manager) findChartURL(name, version, repoURL string, repos map[string]*
 			return
 		}
 	}
-	url, err = repo.FindChartInRepoURL(repoURL, name, version, certFile, keyFile, caFile, m.Getters)
+
+	if _, ok := urls[name]; ok {
+		url = urls[name]
+	} else {
+		url, err = repo.FindChartInRepoURL(repoURL, name, version, certFile, keyFile, caFile, m.Getters)
+	}
```

Comment:
`name` is not globally unique. There could be a chart with the same name in two different repos. And, in a chart there could be multiple versions used. The key lookup needs to deal with this. While somewhat rare, I've encountered what I just described. The cache key needs to account for the repo, name, and version. Just like the `FindChartInRepoURL` function does. This starts in the `resolver.Resolve` function. I point it out here because this is where the impact is felt.

Reply:
@mattfarina hopefully the key including the repo, name, version should suffice

Reply:
If anymore changes are needed just let me know. Hopefully this can get merged before next release